apps/theming/lib/Controller/ThemingController.php

```diff
@@ -372,9 +372,29 @@ public function getStylesheet() {
 	 */
 	public function getManifest($app) {
 		$cacheBusterValue = $this->config->getAppValue('theming', 'cachebuster', '0');
+		if ($app === 'core') {
+			$name = $this->themingDefaults->getName();
+			$short_name = $this->themingDefaults->getName();
+			$start_url = $this->urlGenerator->getBaseUrl();
+			$description = $this->themingDefaults->getSlogan();
+		} else {
+			$info = $this->appManager->getAppInfo($app);
+			$name = $info['name'] . ' - ' . $this->themingDefaults->getName();
+			$short_name = $info['name'];
+			if (strpos($this->request->getRequestUri(), '/index.php/') !== false) {
+				$start_url = $this->urlGenerator->getBaseUrl() . '/index.php' . $this->urlGenerator->linkTo($app, '');
+			} else {
+				$start_url = $this->urlGenerator->getBaseUrl() . $this->urlGenerator->linkTo($app, '');
+			}
```

Comment:
That won't work as linkTo will use the actuall apps directory, which is not what the app route itself will be. E.g. if you have a separate apps2 directory the generated url would be /index.php/apps2/myapp instead of /index.php/apps/myapp.

Also while most frontend apps provide such a route, there is no guarantee that this is the actual entrypoint.

I think the best approach here would be to either just use the current url or get the navigation manager and try to find the entry for the app id, but both also have their limitations.

Reply:
@juliushaertl Thanks for the review!
What about doing it like this:
```suggestion
			if (strpos($this->request->getRequestUri(), '/index.php/') !== false) {
				$start_url = $this->urlGenerator->getBaseUrl() . '/index.php/apps/' . $app . '/';
			} else {
				$start_url = $this->urlGenerator->getBaseUrl() . '/apps/' . $app . '/';
			}
```
I just didn't wanted to hardcode the apps directory but if it is always like this, it maybe makes sense to do so...

Reply:
Either way this doesn't need to match everything as any app is free to overwrite the manifest on itself like e.g. the bookmarks app already does. This PR shall cover most cases (but not all) as it is already a big improvement over how it currently is.

Reply:
@juliushaertl I've done it now this way. please have another look :)

Reply:
What about checking if there is a route with '/' as 'url'?
If there is such a route, then `BASE_URL/apps/APP_ID/` is fine.
Otherwise use the current url.

Reply:
@eneiluj how can I get the currentl url via PHP? This would probably need to be called directly in the php template during loading, no?
`$this->request->getRequestUri()` in this function only returns e.g. `https://192.168.179.192/apps/theming/manifest/mail?v=13`.

Reply:
But as I said before it only shall cover most cases and not all and since the /apps/ directory is the default, I think this is fine because any app that wants to have more complicated routing can and should implement the webmanifest on its own, imho.

Reply:
Or were you talking about something else?

Reply:
I'm saying maybe `BASE_URL/apps/APP_ID/whatever` is the app entry point so `BASE_URL/apps/APP_ID/` leads to nowhere and user will be redirected to NC home page.

As for the custom app directory, I don't think it affects the routing (but I'm not sure, never used it).

Reply:
Ahh okay, no I don't think that this is a problem since all apps that I've tried have the trailing slash in the URL directly when I click on the link in the appmenu before the app is loaded (I think it is the default Nextcloud routing)
Apps that I've tried are those:
![image](https://user-images.githubusercontent.com/42591237/130603689-d8e31b3c-ca41-437f-bf11-8ec34705ae4b.png)

Reply:
It's not about the trailing slash. Some apps might have no route for '/'. Imagine if the contacts app could only be reached at `https://my.nc.org/index.php/apps/contacts/contact-list` and that `https://my.nc.org/index.php/apps/contacts/` would lead to nowhere.

Reply:
> how can I get the currentl url via PHP?

I didn't realize the apps get the manifest with a separate request... An easy solution would have been the referrer header but it's not provided in NC. Then I don't know how to get the current url in this context...:grin:

Reply:
> It's not about the trailing slash. Some apps might have no route for '/'. Imagine if the contacts app could only be reached at https://my.nc.org/index.php/apps/contacts/contact-list and that https://my.nc.org/index.php/apps/contacts/ would lead to nowhere.

If that would be the case the app would also not be reachable via the appmenu since when you click on an app in the appmenu it always opens `BASE_URL/apps/APP_ID/` so you must provide such a route because otherwise your app cannot be opened via the appmenu, as far as I understand it...

Reply:
The navigation item target is defined by the app.
It can be set in info.xml, like that:
https://github.com/eneiluj/cospend-nc/blob/8a4b09298f14bd184fc39452c61ce30069f43591/appinfo/info.xml#L52-L60
or by calling `INavigationManager::add` like that:
https://github.com/eneiluj/integration_openproject/blob/2bc55d2503a6b35e49bcdc089de18fe458ff638a/lib/AppInfo/Application.php#L75-95

It can be whatever we want, even something outside Nextcloud.

Reply:
Okay but as this would be a very specific usecase I think apps that need this should overwrite the webmanifest in this case on their own which the bookmarks app already does, btw.

Reply:
Yeah I agree. I'm fine with `BASE_URL/apps/APP_ID/` since the beginning :grin:. It's better than using NC base url as ` start_url` anyway.